esphome/components/stepper/__init__.py

```diff
@@ -143,6 +145,42 @@ async def stepper_set_speed_to_code(config, action_id, template_arg, args):
     return var
 
 
+@automation.register_action(
+    "stepper.set_acceleration",
+    SetAccelerationAction,
+    cv.Schema(
+        {
+            cv.Required(CONF_ID): cv.use_id(Stepper),
+            cv.Required(CONF_ACCELERATION): cv.templatable(validate_acceleration),
+        }
+    ),
+)
+async def stepper_set_acceleration_to_code(config, action_id, template_arg, args):
+    paren = await cg.get_variable(config[CONF_ID])
+    var = cg.new_Pvariable(action_id, template_arg, paren)
+    template_ = await cg.templatable(config[CONF_ACCELERATION], args, cg.int32)
```

Comment:
Ignore the fact that `speed` is an int above (Or change it too maybe)...
But the config validation and the c++ code expects a float, so the template should allow a float here too. Also below in deceleration

```suggestion
    template_ = await cg.templatable(config[CONF_ACCELERATION], args, cg.float_)
```